### PR TITLE
Validate fed utxos below minimum

### PR DESCRIPTION
**Pull request by jeremy-then**
Renames `PegUtils::isAnyUTXOAmountBelowMinimum` to `PegUtils::allUTXOsToFedAreAboveMinimumPeginValue` method and post RSKIP379 fixes a bug and behaviour.

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java [hunk 1]
```diff
@@ -637,4 +631,328 @@ void test_getTransactionType_flyover_segwit() {
         // Assert
         Assertions.assertEquals(PegTxType.UNKNOWN, pegTxType);
     }
+
+    @Test
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java:635]: Let's move all these tests to a new class `PegUtilsAllUTXOsToFedAreAboveMinimumPeginValueTest`. That way we avoid ending up with a huge test class

**Review by nathanieliov — COMMENTED** [review]
Well done. According to our discussion, please consider adding the following tests:

```
- Before RSKIP379
	valid pegin - all fed utxos above the minimum
		only one output to active fed
 		many outputs to active fed and retiring fed equal to minimum
 		one output equal to minimum to retiring fed
 		**none output to any fed but outputs to unknown addresses *************
 		invalid pegin - some fed utxos above minimum and other below minimum
 		only one output below minimum to active fed
 		many outputs to active fed and  some below minimum to retiring fed
 		one output below minimum to retiring fed
- After RSKIP379
	valid pegin - all fed utxos above minimum
		only one output to active fed
		many outputs to active fed and retiring fed equal to minimum
		one output equal to minimum to retiring fed
		invalid pegin - some fed utxos above minimum and other below minimum
		only one output below minimum to active fed
		many outputs to active fed and  some below minimum to retiring fed
		one output below minimum to retiring fed
		**none output to any fed but outputs to unknown addresses *************
```

File: rskj-core/src/main/java/co/rsk/peg/PegUtils.java [hunk 2]
```diff
@@ -37,7 +37,33 @@ public static PegTxType getTransactionTypeUsingPegoutIndex(
         }
     }
 
-    public static boolean isAnyUTXOAmountBelowMinimum(Coin minimumPegInTxValue, BtcTransaction btcTx, Wallet wallet) {
-        return PegUtilsLegacy.isAnyUTXOAmountBelowMinimum(minimumPegInTxValue, btcTx, wallet);
+    /**
+     * Checks if all utxos sent to the <b>fedWallet</b> are equal or above the minimum pegin value.
+     * If no utxo is sent to the <b>fedWallet </b>or there is at least 1 utxo sent to the <b>fedWallet</b> that is below minimum,
+     * this method will return false. It will return true when all utxos sent to the <b>fedWallet</b> are above the minimum pegin value.
+     * Before RSKIP379, this method calls PegUtilsLegacy::isAnyUTXOAmountBelowMinimum, which has a bug and might return some unexpected values.
```
[the remark this answers is not in the document]
Comment (jeremy-then) [reply rskj-core/src/main/java/co/rsk/peg/PegUtils.java]: Updated!

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java [hunk 3]
```diff
@@ -1,12 +1,6 @@
 package co.rsk.peg;
 
-import co.rsk.bitcoinj.core.Address;
-import co.rsk.bitcoinj.core.BtcECKey;
-import co.rsk.bitcoinj.core.BtcTransaction;
-import co.rsk.bitcoinj.core.Coin;
-import co.rsk.bitcoinj.core.Context;
-import co.rsk.bitcoinj.core.NetworkParameters;
-import co.rsk.bitcoinj.core.Sha256Hash;
+import co.rsk.bitcoinj.core.*;
```
[the remark this answers is not in the document]
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java]: Updated! But using the `.*` does not actually import all the classes in that package. The compiler is smart enough to remove all the things that are not being used and actually inserts all the imports with the fully qualified name for us. I've tried this by compiling a class using `.*` and then decompiling it and the everything not used is removed and only classes that are being used are imported.

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java (hunk 1 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java]: Updated!

**Comment by nathanieliov** [comment]
Closing this PR  because these changes were merged and will be reviewed in PR #2177.

**Pull request closed without merging**